### PR TITLE
Define default URI template

### LINKED ISSUE #46: Consider adding a default template
Align with CONNECT-UDP to use the well-known default template

### PR DESCRIPTION
**Pull request by tfpauly**
Closes #46

### CODE REVIEW

File: draft-ietf-masque-connect-ip.md [hunk 1]
```diff
@@ -131,7 +132,9 @@ The following requirements apply to the URI Template:
 * All template variables MUST be within the path or query components of the URI.
 
 * The URI template MAY contain the two variables "target" and "ipproto" and MAY
-  contain other variables.
+  contain other variables. If the "target" and "ipproto" variables are included
+  as path components, their values MUST NOT be empty, but instead use "\*" to
+  indicate wildcard or no-preference values.
```
Comment (DavidSchinazi) [inline draft-ietf-masque-connect-ip.md:137]:
```suggestion
  contain other variables. If the "target" or "ipproto" variables are included,
  their values MUST NOT be empty. Clients can instead use "\*" to indicate
  wildcard or no-preference values, see {{scope}}..
```

File: draft-ietf-masque-connect-ip.md [hunk 2]
```diff
@@ -148,6 +151,14 @@ If a client detects that any of the requirements above are not met by a URI
 Template, the client MUST reject its configuration and abort the request without
 sending it to the IP proxy.
 
+As with CONNECT-UDP, some client configurations for CONNECT-IP proxies will only
+allow the user to configure the proxy host and proxy port. Clients with such limitations
+MAY attempt to access IP proxying capabilities using the default template, which is
+defined as: "https://$PROXY_HOST:$PROXY_PORT/.well-known/masque/udp/{target}/{ipproto}/",
```
Comment (DavidSchinazi) [inline draft-ietf-masque-connect-ip.md:157]:
```suggestion
defined as: "https://$PROXY_HOST:$PROXY_PORT/.well-known/masque/ip/{target}/{ipproto}/",
```

File: draft-ietf-masque-connect-ip.md [hunk 3]
```diff
@@ -826,6 +837,20 @@ References:
...
 {: spacing="compact"}
 
+## Updates to masque Well-Known URI {#iana-uri}
+
+This document will request IANA to update the entry for the "masque"
+URI suffix in the "Well-Known URIs" registry maintained at
+<[](https://www.iana.org/assignments/well-known-uris)>.
+
+IANA is requested to update the "Reference" field to include this
+document.
```
Comment (DavidSchinazi) [inline draft-ietf-masque-connect-ip.md:847]:
```suggestion
document in addition to previous values from that field.
```

File: draft-ietf-masque-connect-ip.md [hunk 4]
```diff
@@ -113,6 +113,7 @@ works to exclude it in the syntax.
 Examples are shown below:
 
 ~~~
+https://example.org/masque/{target}/{ipproto}/
```
Comment (DavidSchinazi) [inline draft-ietf-masque-connect-ip.md:116]: Can you also add the default template to the example?

**Review by DavidSchinazi — APPROVED** [review]
Overall looks good modulo a bunch of minor comments

File: draft-ietf-masque-connect-ip.md (hunk 4 above)
Comment (tfpauly) [reply draft-ietf-masque-connect-ip.md:116]: Oh whoops, I should have done it with this one

Comment (tfpauly) [reply draft-ietf-masque-connect-ip.md:116]:
```suggestion
https://example.org/.well-known/masque/ip/{target}/{ipproto}/
```

File: draft-ietf-masque-connect-ip.md [hunk 5]
```diff
@@ -131,7 +132,9 @@ The following requirements apply to the URI Template:
 * All template variables MUST be within the path or query components of the URI.
 
 * The URI template MAY contain the two variables "target" and "ipproto" and MAY
-  contain other variables.
+  contain other variables. If the "target" or "ipproto" variables are included,
+  their values MUST NOT be empty. Clients can instead use "\*" to indicate
+  wildcard or no-preference values, see {{scope}}..
```
Comment (DavidSchinazi) [inline draft-ietf-masque-connect-ip.md:137]:
```suggestion
  wildcard or no-preference values, see {{scope}}.
```

**Review by DavidSchinazi — APPROVED** [review]

**Review by achernya — APPROVED** [review]

**Review by gloinul — APPROVED** [review]

**Review by mirjak — APPROVED** [review]

File: draft-ietf-masque-connect-ip.md [hunk 6]
```diff
@@ -826,6 +837,20 @@ References:
...
+## Updates to masque Well-Known URI {#iana-uri}
+
+This document will request IANA to update the entry for the "masque"
+URI suffix in the "Well-Known URIs" registry maintained at
+<[](https://www.iana.org/assignments/well-known-uris)>.
+
+IANA is requested to update the "Reference" field to include this
+document in addition to previous values from that field.
+
+IANA is request to add the following sentence to the "Related Information"
```
Comment (martinthomson) [inline draft-ietf-masque-connect-ip.md:849]:
```suggestion
IANA is requested to add the following sentence to the "Related Information"
```

**Review by martinthomson — APPROVED** [review]

**Pull request merged**